libs/api/transaction/data-access/src/lib/api-transaction-data-access.service.ts

```diff
@@ -149,10 +150,18 @@ export class ApiTransactionDataAccessService implements OnModuleInit {
     return { lamports } as MinimumRentExemptionBalanceResponse
   }
 
-  async makeTransfer(input: MakeTransferRequest): Promise<AppTransactionWithErrors> {
+  async makeTransfer(req: Request, input: MakeTransferRequest): Promise<AppTransactionWithErrors> {
     const { appEnv, appKey } = await this.data.getAppEnvironment(input.environment, input.index)
     this.makeTransferRequestCounter.add(1, { appKey })
 
+    if (appEnv?.ipsAllowed.length > 0 && !appEnv?.ipsAllowed.includes(req.ip)) {
```

Comment:
When using `request-ip` as middleware, the property we are look at is no longer `req.ip` but `req.clientIp` - [reference here](https://github.com/pbojinov/request-ip/blob/master/src/index.js#L183).